tests/shared/core/training_data/test_visualization.py

```diff
@@ -188,3 +188,21 @@ def test_story_visualization_with_merging(domain: Domain):
     assert 15 < len(generated_graph.nodes()) < 33
 
     assert 20 < len(generated_graph.edges()) < 33
+
+
+def test_remove_auxiliary_nodes():
+    # Create a sample graph
+    graph = nx.MultiDiGraph()
+    graph.add_nodes_from([-2, -1, 0, 1, 2, 3, 4, 5])
+    graph.add_edges_from([(-2, 0), (-1, 0), (0, 1), (1, 2), (2, 3), (3, 4), (4, 5)])
+
+    # Call the method to remove auxiliary nodes
+    visualization._remove_auxiliary_nodes(graph, 3)
```

Comment:
The test fails with `networkx.exception.NetworkXError: The node -1 is not in the digraph.` 🤔  didn't investigate much please explain why it was happening.

Reply:
Hello @Urkem,
I have implemented two example test cases for the same. Could you please review them whenever you have a moment? Let me know if you are still facing any issues while testing. 
Thank you!

Reply:
@arjun-234 Thanks for the new test case, they both pass now 👍